src/node.h

```diff
@@ -148,6 +148,77 @@ typedef intptr_t ssize_t;
 
 namespace node {
 
+  enum NodeInstanceType { MAIN, WORKER };
+
+  class NodeInstanceData {
+    public:
+
+
+      NodeInstanceData(NodeInstanceType node_instance_type,
+                       uv_loop_t* loop,
+                       int argc,
+                       const char** argv,
+                       int exec_argc,
+                       const char** exec_argv) :
+          node_instance_type_(node_instance_type),
+          loop_(loop),
+          argc_(argc),
+          argv_(argv),
+          exec_argc_(exec_argc),
+          exec_argv_(exec_argv) {
+      }
+
+      inline uv_loop_t* loop() {
+        assert(loop_);
```

Comment:
Please use CHECK_EQ or CHECK_NE.  assert, ASSERT and CHECK are really only intended for true booleans (e.g. `CHECK(callback->IsFunction())`.)